compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -1568,6 +1569,28 @@ class Objects(using Context @constructorOnly):
         throw new Exception("unexpected type: " + tp + ", Trace:\n" + Trace.show)
   }
 
+  /** Widen the escaped value (a method argument or rhs of an assignment)
+   *
+   *  The default widening is 1 for most values, 2 for function values.
+   *  User-specified widening annotations are repected.
+   */
+  def widenEscapedValue(value: Value, expr: Tree): Contextual[Value] =
```

Comment:
I'd suggest renaming `expr` to something more specific such as `annotatedTree` or `annotationHost`.